test/runtests.py

```diff
@@ -7,114 +7,208 @@
+# records pass_count/fail_count
+class PassFailCount(object):
+    def __init__(self):
+        self.pass_count = 0
+        self.fail_count = 0
+
+    def __str__(self):
+        return 'passed {}, failed {}'.format(self.pass_count, self.fail_count)
+
+# records total and individual folder's pass_count/fail_count
+class TestResult(PassFailCount):
+    def __init__(self):
+        super(self.__class__, self).__init__()
+        self.folders = {}
 
-        print "-?, --help    : Show help\n"
+    def _get_folder_result(self, folder):
+        r = self.folders.get(folder)
+        if not r:
+            r = PassFailCount()
+            self.folders[folder] = r
+        return r
 
-        print "Samples:"
-        print "test only Array:"
-        print "\t./test.py Array\n"
-        
-        print "test a single file:"
-        print "\t./test.py Basics/hello.js\n"
+    def log(self, filename=None, folder=None, fail=False):
+        if not folder:
+            folder = os.path.basename(os.path.dirname(filename))
+        r = self._get_folder_result(folder)
+        if fail:
+            r.fail_count += 1
+            self.fail_count += 1
+        else:
+            r.pass_count += 1
+            self.pass_count += 1
 
-        print "test all folders:"
-        print "\t./test.py"
-        sys.exit(0)
-    test_all = None
+test_result = TestResult()
 
-test_dirs=['']
-if test_all:
-    test_dirs = os.listdir(test_root)
-else:
-    test_dirs[0] = sys.argv[1]
 
 def show_failed(filename, output, exit_code, expected_output):
-    print "\nFailed ->", filename
+    print("\nFailed -> {}".format(filename))
     if expected_output == None:
-        print "\nOutput:"
-        print "----------------------------"
-        print output
-        print "----------------------------"
+        print("\nOutput:")
+        print("----------------------------")
+        print(output)
+        print("----------------------------")
     else:
         lst_output = output.split('\n')
         lst_expected = expected_output.split('\n')
         ln = min(len(lst_output), len(lst_expected))
         for i in range(0, ln):
             if lst_output[i] != lst_expected[i]:
-                print "Output: (at line " + str(i) + ")" 
-                print "----------------------------"
-                print lst_output[i]
-                print "----------------------------"
-                print "Expected Output:"
-                print "----------------------------"
-                print lst_expected[i]
-                print "----------------------------"
+                print("Output: (at line " + str(i) + ")")
+                print("----------------------------")
+                print(lst_output[i])
+                print("----------------------------")
+                print("Expected Output:")
+                print("----------------------------")
+                print(lst_expected[i])
+                print("----------------------------")
```

Comment:
Are you planning on using `from __future__ import print_function`?

Otherwise using what looks like function syntax with the print function is actually printing a tuple with one element.